pkg/analyze/host_sysctl.go

```diff
@@ -0,0 +1,101 @@
+package analyzer
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/replicatedhq/troubleshoot/pkg/collect"
+)
+
+// Ensure `AnalyzeHostSysctl` implements `HostAnalyzer` interface at compile time.
+var _ HostAnalyzer = (*AnalyzeHostSysctl)(nil)
+
+type AnalyzeHostSysctl struct {
+	hostAnalyzer *troubleshootv1beta2.HostSysctlAnalyze
+}
+
+func (a *AnalyzeHostSysctl) Title() string {
+	return hostAnalyzerTitleOrDefault(a.hostAnalyzer.AnalyzeMeta, "Sysctl")
+}
+
+func (a *AnalyzeHostSysctl) IsExcluded() (bool, error) {
+	return isExcluded(a.hostAnalyzer.Exclude)
+}
+
+func (a *AnalyzeHostSysctl) Analyze(
+	getCollectedFileContents func(string) ([]byte, error), findFiles getChildCollectedFileContents,
+) ([]*AnalyzeResult, error) {
+	result := AnalyzeResult{Title: a.Title()}
+
+	// Use the generic function to collect both local and remote data
+	collectedContents, err := retrieveCollectedContents(
+		getCollectedFileContents,
+		collect.HostSysctlPath,     // Local path
+		collect.NodeInfoBaseDir,    // Remote base directory
+		collect.HostSysctlFileName, // Remote file name
+	)
+	if err != nil {
+		return []*AnalyzeResult{&result}, err
+	}
+
+	results, err := analyzeHostCollectorResults(collectedContents, a.hostAnalyzer.Outcomes, a.CheckCondition, a.Title())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to analyze sysctl output")
+	}
+
+	return results, nil
+}
+
+// checkCondition checks the condition of the when clause
+func (a *AnalyzeHostSysctl) CheckCondition(when string, data []byte) (bool, error) {
+
+	sysctl := map[string]string{}
+	if err := json.Unmarshal(data, &sysctl); err != nil {
+		return false, errors.Wrap(err, "failed to unmarshal data")
+	}
+
+	// <1:key> <2:operator> <3:value>
+	matches := sysctlWhenRX.FindStringSubmatch(when)
+	if len(matches) < 4 {
+		return false, fmt.Errorf("expected 3 parts in when %q", when)
+	}
+
+	param := matches[1]
+	operator := matches[2]
```

Comment:
You can use [ParseComparisonOperator](https://github.com/replicatedhq/troubleshoot/blob/2f62240ed3390977375b15b23b05586b3f6eb2be/pkg/analyze/comparison.go#L17) to parse operator comparator

Reply:
Done via - https://github.com/replicatedhq/troubleshoot/pull/1681/commits/258d8831baf9e46c48b819ad1b1dc1ab779a909b